added en.po file

patogit opened the pull request:
don't know why it wasn't there. Just copied the msgid to the msgstr.

p4u commented:
I might not understand something but  why is a po English file needed if Enflish is the main/reference language which is written in the LUA code.

https://github.com/libremesh/lime-packages/blob/develop/packages/lime-webui/luasrc/model/essentials.lua

patogit commented:
Without a .PO or .POT file in English, the translation interface won't work.  We need that file in order for the translation platform to show the strings that we will translate, i.e. the source file.

But maybe me adding it by hand isn't the proper way? Maybe it's supposed to be done with a script, in order to keep everything up to date?

patogit commented:
See quoted request for this file at https://github.com/libremesh/lime-packages/issues/257#issuecomment-367206873

patogit commented:
@p4u , any action on this? It's one of the last pieces we're waiting on in order to set up the space at TranslateWiki.

Nikerabbit commented:
I am not sure whether this file will work. It seems to lack the normal gettext headers. Compare with [this](https://github.com/mantisbt-plugins/codev/blob/master/i18n/locale/codevtt.pot#L6-L18) for example (also msgstr is empty in pot files).

patogit commented:
I just updated to repo. Deleted the po/en/lime-webui.po and added po/lime-webui.pot with headers as indicated by @Nikerabbit . (Sorry for the delay, my work sometimes takes me out of Internet connectivity.)

patogit commented:
Now I'm reading the docs on how to generate the POT file autmatically... https://gluon.readthedocs.io/en/v2017.1.x/dev/web/i18n.html#adding-translation-templates-to-gluon-packages

patogit commented:
updated strings using script, and then re-inserted the headers. I wonder how to make the script insert the headers.

altergui commented on packages/lime-webui/po/lime-webui.pot:
```diff
@@ -1,16 +1,21 @@
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
-"Language: en_US\n"
+"Language: it_IT\n"
```
why it_IT?

patogit replied:
It's just a template. That language code will get changed in each real translation file. it_IT is just an example.

altergui commented:
err, i mean the three commits can be squashed together (the first two commits are pointless)

patogit commented:
I don't know how to squash those three commits, and the tutorials I'm trying aren't working.

Nikerabbit commented on packages/lime-webui/po/lime-webui.pot:
```diff
@@ -0,0 +1,248 @@
+msgid ""
```
Pot files should start with `#, fuzzy`

patogit replied:
Ok, I added that in a commit just now.

Nikerabbit commented:
One can "Squash and merge" with GitHub directly.

Also, can you clarify whether this file is generated by script or manually? I think there must be a script that keeps this file up to date when strings are changed or you won't be getting translations for the new strings.

patogit commented:
I don't see any "Squash & merge" button on GitHub. Maybe it's only for repo owners? Or maybe it's an option at the moment of merging the pull request? This makes sense to me.

I squashed the commits on my local repo, and git told me to pull first and then push, so I did, and we'll see if that worked.

patogit commented:
> Also, can you clarify whether this file is generated by script or manually? I think there must be a script that keeps this file up to date when strings are changed or you won't be getting translations for the new strings.

The script to keep the file up to date isn't automated, and it doesn't insert the headers nor the `#, fuzzy` line. To generate the file, I ran the script, and then I added the headers and the `#, fuzzy` line manually.

I looked at the script to see if I could add the headers and fuzzy, but I don't understand the script. As for automating it when commits are made, maybe there's a way to do that with GitHub web hooks? Or maybe there's a different script that meets our needs better?

altergui commented:
Nice, this squashed commit is just what i had in mind
29ec735c7e
contgratulations!
problem is, you *merged* that commit with the "messy" branch (03f8cc2974), resulting in 2861f04488, which is even messier than the original branch

using git via command line, this is what i would do:
`git checkout -B hotfix/create-en-po-file 29ec735c7e`
`git push -f git://github.com/patogit/lime-packages.git hotfix/create-en-po-file`

that will force a non-fastforward push to your repo, and you'll lose the original 4 commits, but that's exactly what we want :)

altergui commented:
i don't have enough experience with github to say if it will allow you to do that. in case it doesn't, simply push that nice commit to a new branch and open a new pull request :(

patogit commented:
> using git via command line, this is what i would do:
`git checkout -B hotfix/create-en-po-file 29ec735c7e`
`git push -f git://github.com/patogit/lime-packages.git hotfix/create-en-po-file`

Done. The only thing I had to change was the protocol of the second command, from `git://` to `https://`.

Does it look the way we want it now?

patogit commented:
It looks clean here, https://github.com/patogit/lime-packages/commits/hotfix/create-en-po-file , only one commit. I'm not sure what I should have done differently in the first place, but hey, it's done now.

altergui commented:
i can't find a way in github to see graph the messy branch.
using gitk (`apt install gitk`), it's evident.
but yes, now it's clean :)
